### ISSUE TITLE
More statistics (suggestion)

### ISSUE THREAD

**Issue by josealberto4444** [SamAmco/track-and-graph#102]
It would be nice if "average time between" stat also told the standard deviation, so one can distinguish if that average is accurate or not.

Also, for numerical data, it would be nice to show:
- Min-max range
- First-last range
- Number of entries
- Sum
- Mean/average
- Standard deviation
- Median

I had, long ago, an app that did all these calculations and I miss it a lot. I think this is the best similar app in f-droid, so maybe it can be improved even more to show those quantities. Thanks for the great app and the hard work of all contributors!

**Comment by SamAmco** [comment]
These are some nice ideas, thank you.

**Comment by stheid** [comment]
I think this statistics are different than graphs, as a graph in the app can be fueled by multiple data series. However such a statistic is bound to a dataseries. Therefore it might make sense to calculate all of those statistics and show it in the FragmentFeatureHistory

**Comment by SamAmco** [comment]
Actually I do agree with the idea of having these as statistics because one of the major updates i would like to add in the long run is "[functions](https://github.com/SamAmco/track-and-graph/issues/69)" which would make these more useful as you could add cards for statistics based on derivatives of multiple data sources.

**Comment by stheid** [comment]
But than it should be easy to generate such functions (this also counts for simple graphs) directly from the tracker. Its the source of the visualization, therefore one should not need to do 5 clicks for creating simple uncombined statistics and graphs.

**Comment by SamAmco** [comment]
Yeah perhaps this is fair enough and maybe it makes sense to have some simple statistics accessible from the history view. At the same time though the idea is that the tracking side is agnostic about the type of data being tracked. A common one here is the fact that you can track data that does not have a natural order like a multiple choice tracker with choices (red, green, blue) etc. Of course multiple choice answers are assigned a value like 0, 1, 2 etc but it makes no sense to draw a line graph of these numbers or generate an average of them. You would need to think carefully about what kind of statistics are consistently relevant and how to avoid confusion for the user here. I have been thinking about getting rid of track groups and graph groups all together and just having groups that can contain trackers, or graphs, or even other groups. This way you could have a tracker and a graph/statistic in the same group and see the data updated immediately. This would probably be a hefty refactor though.

**Comment by stheid** [comment]
agree, but i think the type of data already determines quite a lot about useful statistics:

for all data you can have: number of datapoints, time domain (first, last datapoint)

numerical and time data:  can have min,max,median mean, std...
ordinal data, can have median
categorical data can have mode


I think it would anyway be valuable to distingish categorical and ordinal data in the UI (which is not done right now) as it allows for more sensible "default plots"

**Comment by SamAmco** [comment]
@stheid I think as a first implementation we should avoid adding the distinction of categorical vs ordinal data in the set up phase and treat this as a separate issue. One concern I always have with new features is maintaining the simplicity for new users sake. If you want to tackle that I think it might be best discussed in a new PR. However you do make a good point there is still plenty of data which is almost always relevant. I'm happy for you to implement this if you would like. I would suggest the best solution is to improve the dialog that comes up when you press the information button in the top right of the history screen. Important to note though is that features can have descriptions of unbounded length, so this must be scrollable.

**Comment by MikeGrace** [comment]
I'll second the desire for showing various calculations for data. I'm particularly interested in Sum. If I'm tracking how many miles I run each day, it would be great to be able to show a sum for basic time frames like year, month, week.

**Comment by biosaat** [comment]
I would like to use the app to track my monthly gas and current meter reading. Therefore the function **difference** would be helpful: one could just track the current meter reading and plot the differences in a graph.
